### PR TITLE
add in featurizers entry point tests

### PR DESCRIPTION
**Pull request by angryjinyan**

### CODE REVIEW

File: test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs [hunk 1]
```diff
@@ -39,34 +39,164 @@ namespace Microsoft.ML.RunTests
...
-
-
-        private class DateTimeInput
+        public TestFeaturizersEntryPoints(ITestOutputHelper output) : base(output)
         {
-            public long date;
+            Env.ComponentCatalog.RegisterAssembly(typeof(RollingWindowEstimator).Assembly);
+            Env.ComponentCatalog.RegisterAssembly(typeof(LagLeadOperatorEstimator).Assembly);
+            Env.ComponentCatalog.RegisterAssembly(typeof(ForecastingPivotFeaturizerEstimator).Assembly);
+            Env.ComponentCatalog.RegisterAssembly(typeof(ShortGrainDropperEstimator).Assembly);
```
Comment (michaelgsharp) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs]: You only need to do this once with one class. It is registering the Assembly and they are all in the same assembly.

Comment (angryjinyan) [reply test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs]: Good to know! I have changed it to just register one class.

File: test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs [hunk 2]
```diff
@@ -39,34 +39,164 @@ namespace Microsoft.ML.RunTests
...
+            while (cursor.MoveNext())
+            {
+                getter(ref buffer);
+                var bufferValues = buffer.GetValues();
+
+                Assert.Equal(expectedOutput[index].Length, bufferValues.Length);
+                Assert.Equal(expectedOutput[index++][0], bufferValues[0]);
+            }
+            // TODO: Uncomment when featurizer fixed.
+            //TestEstimatorCore(pipeline, data);
```
Comment (michaelgsharp) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs]: You can remove these 2 comments/lines in all the tests. They wont apply to your code.

File: test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs [hunk 3]
```diff
@@ -39,34 +39,164 @@ namespace Microsoft.ML.RunTests
...
+            Done();
+        }
+        [Fact]
```
Comment (michaelgsharp) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs:171]: Make sure you have spaces between the `}` and `[Fact]` for readability.

File: test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs [hunk 4]
```diff
@@ -39,34 +39,160 @@ namespace Microsoft.ML.RunTests
...
+
+            var runner = new GraphRunner(Env, graph[FieldNames.Nodes] as JArray);
+
+            runner.SetInput("data", data);
+            runner.RunAll();
+
+            var output = runner.GetOutput<IDataView>("outputData");
+            var schema = output.Schema;
+
+            var addedColumn = schema["Target_Mean_Hor1_MinWin1_MaxWin1"];
```
Comment (ganik) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs:92]: >Target_Mean_Hor1_MinWin1_MaxWin1 [](start = 38, length = 32)

Where does this strange column names come from?

File: test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs [hunk 5]
```diff
@@ -78,41 +204,216 @@ public void DateTime()
...
+
+            var addedColumn = schema["Target_Mean_Hor4_MinWin2_MaxWin3"];
+            var columnType = addedColumn.Type as VectorDataViewType;
+
+            // Make sure the type and schema of the column are correct.
+            Assert.NotNull(columnType);
+            Assert.True(columnType.IsKnownSize);
+            Assert.True(columnType.Dimensions.Length == 2);
+            Assert.True(columnType.Dimensions[0] == 1);
+            Assert.True(columnType.Dimensions[1] == 4);
```
Comment (ganik) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs:290]: >columnType.Dimensions[1] == 4) [](start = 24, length = 30)

slot is a vector itself?

File: test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs [hunk 6]
```diff
@@ -39,34 +39,160 @@ namespace Microsoft.ML.RunTests
...
+                'Nodes':
+                [{
+                    'Name': 'Transforms.AnalyticalRollingWindow',
+                    'Inputs': {
+                            'GrainColumns': ['Grain'],
+                            'TargetColumn' : 'Target',
+                            'Data' : '$data',
+                            'Horizon': 1,
+                            'MaxWindowSize' : 1,
+                            'MinWindowSize' : 1,
```
Comment (ganik) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs:70]: Check:
1. Erroneous input (0, -1)
2. Check big numbers
3. Check nulls
4. Omit fields

File: test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs (hunk 4 above)
Comment (angryjinyan) [reply test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs:92]: This is a column added by analytical rolling window and the column name consists of the input parameters like horizon, calculation type, min and max window size.

File: test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs (hunk 5 above)
Comment (angryjinyan) [reply test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs:290]: Yeah.

File: test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs (hunk 6 above)
Comment (angryjinyan) [reply test/Microsoft.ML.Core.Tests/UnitTests/TestFeaturizersEntryPoints.cs:70]: Erroneous input and nulls will be caught by the compiler with error messages `Field ... is required but not set` and `Values was either too large or too small for a type`. Tests for big numbers and omit fields are added.

**Pull request merged**